registry/auth.go

```diff
@@ -74,6 +74,12 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorized() bool {
```

Comment:
@stevvooe we will need to find a solution for this for the refactor anyway, we should determine what the set of API requests should be then make sure we are following that. Ideally we do `GET /v2/`, optionally get token, then push data or head check.  Ideally we could always do a head check but doing an extra digest calculation is costly.